internal/crossdock/internal/header.go

```diff
@@ -20,7 +20,10 @@
 
 package internal
 
+import "go.uber.org/yarpc/transport/tchannel"
+
 // RemoveVariableMapKeys removes any headers that might have been added by tracing
 func RemoveVariableMapKeys(headers map[string]string) {
 	delete(headers, "$tracing$uber-trace-id")
+	delete(headers, tchannel.ServiceHeaderKey)
```

Comment:
Let's change the name of this function and the comment since it's not specific to tracing anymore.

Also, let's add a comment as to why we need to remove the service header here.

Reply:
Comments changed.